src/main/java/de/hysky/skyblocker/utils/datafixer/ItemStackComponentizationFixer.java

```diff
@@ -48,15 +49,16 @@ public static String componentsAsString(ItemStack stack) {
 		RegistryOps<NbtElement> nbtRegistryOps = getRegistryLookup().getOps(NbtOps.INSTANCE);
 
 		return Arrays.toString(stack.getComponentChanges().entrySet().stream().map(entry -> {
-			@SuppressWarnings("unchecked")
-			ComponentType<Object> dataComponentType = (ComponentType<Object>) entry.getKey();
-			Identifier componentId = Registries.DATA_COMPONENT_TYPE.getId(dataComponentType);
-			Optional<NbtElement> encodedComponent = dataComponentType.getCodec().encodeStart(nbtRegistryOps, entry.getValue().orElseThrow()).result();
+			ComponentType<?> componentType = entry.getKey();
+			Identifier componentId = Registries.DATA_COMPONENT_TYPE.getId(componentType);
+			if (componentId == null) return null;
 
-			if (componentId == null || encodedComponent.isEmpty()) {
-				return null;
-			}
+			Optional<?> component = entry.getValue();
+			if (component.isEmpty()) return "!" + componentId;
```

Comment:
I don't think the game can parse this? what are these changes for

Reply:
I copied this from `ItemStackArgument#asString`.